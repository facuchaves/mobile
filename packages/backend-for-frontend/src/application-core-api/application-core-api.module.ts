import { Module, HttpModule } from '@nestjs/common'
import { ApplicationCoreApiService } from './application-core-api.service'
import { CoreApiModule } from '../core-api/core-api.module'
import { ConfigModule } from '../config/config.module'

@Module({
  imports: [ConfigModule, HttpModule, CoreApiModule],
  providers: [ApplicationCoreApiService],
  exports: [ApplicationCoreApiService],
})
export class ApplicationCoreApiModule {}
