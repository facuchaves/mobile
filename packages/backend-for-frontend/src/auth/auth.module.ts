import { Module } from '@nestjs/common'
import { JwtModule } from '@nestjs/jwt'
import { ConfigModule } from '../config/config.module'
import { ConfigService } from '../config/config.service'

@Module({
  imports: [
    ConfigModule,
    JwtModule.registerAsync({
      useFactory: (configService: ConfigService) => ({
        secret: configService.getJwtSecret(),
        signOptions: {
          // TODO: Por ahora JWT se usa solamente para la sesión
          // A futuro deberíamos pasarle este parámetro cuando firmamos el token
          expiresIn: 3600 * 24 * 30, // 30 dias
        },
      }),
      inject: [ConfigService],
      imports: [ConfigModule],
    }),
  ],
  exports: [JwtModule],
})
export class AuthModule {}
