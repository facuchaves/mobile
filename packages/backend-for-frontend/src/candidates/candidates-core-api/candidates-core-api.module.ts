import { Module, HttpModule } from '@nestjs/common'
import { CandidatesCoreApiService } from './candidates-core-api.service'
import { CoreApiModule } from '../../core-api/core-api.module'
import { ConfigModule } from '../../config/config.module'

@Module({
  imports: [ConfigModule, HttpModule, CoreApiModule],
  providers: [CandidatesCoreApiService],
  exports: [CandidatesCoreApiService],
})
export class CandidatesCoreApiModule {}
