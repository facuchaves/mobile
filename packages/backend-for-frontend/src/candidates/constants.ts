export const PREFIX_CURRICULUM = 'datosCurriculum'