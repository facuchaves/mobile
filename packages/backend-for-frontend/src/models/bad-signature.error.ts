export default class BadSignatureError extends Error {
  constructor(message) {
    super(message)
  }
}
