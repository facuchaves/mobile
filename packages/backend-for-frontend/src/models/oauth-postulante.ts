export default class OauthPostulante {
  accessToken: string
  tokenType: string
  expiresIn: number
  expirationDate: Date
  refreshToken: string

  constructor(oauthJson) {
    this.accessToken = oauthJson.access_token
    this.tokenType = oauthJson.token_type
    this.expiresIn = oauthJson.expires_in
    this.expirationDate = oauthJson.expire_date
    this.refreshToken = oauthJson.refresh_token
  }

  toString() {
    return `AccessToken : ${this.accessToken} \n 
                TokenType : ${this.tokenType} \n 
                ExpiresIn : ${this.expiresIn} \n 
                refreshToken : ${this.refreshToken} \n
                expirationDate : ${this.expirationDate} \n `
  }
}
