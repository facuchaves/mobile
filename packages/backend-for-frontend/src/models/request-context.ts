import { Site } from '../interface/site.interface'
import Oauth from './oauth'
import { Logger } from '../logger/logger.service'
import { isEmpty } from 'lodash'

export class RequestContext {
  site: Site
  oauthApplication: Oauth
  currentOauth: Oauth
  oauthPostulante: Oauth
  logger: Logger
  sessionId: any

  isCandidateLogged(): boolean {
    return !isEmpty(this.oauthPostulante)
  }
}
