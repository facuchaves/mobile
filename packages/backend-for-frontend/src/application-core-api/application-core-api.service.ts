import { Injectable } from '@nestjs/common'
import { AxiosRequestConfig, AxiosResponse } from 'axios'
import { Observable } from 'rxjs'
import { RequestContext } from '../models/request-context'
import { CoreApiService } from '../core-api/core-api.service'

/**
 * Clase que wrapea el CoreApiService para configurarle antes el ouath que debe utilizar
 */
@Injectable()
export class ApplicationCoreApiService {
  constructor(private readonly coreApiService: CoreApiService) {}

  /**
   * Metodo que setea el oauth de candidates en el context
   * @param context
   */
  populateOauth(context: RequestContext) {
    context.currentOauth = context.oauthApplication
  }

  get<T = any>(context: RequestContext, url: string, config?: AxiosRequestConfig): Promise<T> {
    this.populateOauth(context)
    return this.coreApiService.get(context, url, config)
  }

  delete<T = any>(context: RequestContext, url: string, config?: AxiosRequestConfig): Promise<T> {
    this.populateOauth(context)
    return this.coreApiService.delete(context, url, config)
  }

  post<T = any>(context: RequestContext, url: string, data?: any, config?: AxiosRequestConfig): Promise<T> {
    this.populateOauth(context)
    return this.coreApiService.post(context, url, data, config)
  }

  put<T = any>(context: RequestContext, url: string, data?: any, config?: AxiosRequestConfig): Promise<T> {
    this.populateOauth(context)
    return this.coreApiService.put(context, url, data, config)
  }
}
