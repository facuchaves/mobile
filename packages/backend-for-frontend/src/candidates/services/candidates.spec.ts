import { RequestContext } from '../../models/request-context'
import { TestingModule, Test } from '@nestjs/testing'
import { CandidatesService } from './candidates.service'
import { RedisModule } from '../../cache/cache-redis/redis.module'
import { CandidatesCoreApiModule } from '../../candidates/candidates-core-api/candidates-core-api.module'
import { AuthModule } from '../../auth/auth.module'
import { ApplicationCoreApiModule } from '../../application-core-api/application-core-api.module'

describe('Candidates ', () => {
  let service

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      imports: [RedisModule, CandidatesCoreApiModule, AuthModule, ApplicationCoreApiModule],
      providers: [CandidatesService],
    }).compile()

    service = module.get<CandidatesService>(CandidatesService)
  })

  it('should completar req con solo nombre', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { nombre: 'Nombre modificado' })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo apellido', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      apellido: 'Apellido modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo email', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { email: 'Email modificado' })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo paisNacimientoId', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { paisNacimientoId: 2 })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo celularNumero', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      celularNumero: 'CelularNumero modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo celularPrefijo', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      celularPrefijo: 'CelularPrefijo modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo discapacidadDetalle', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      discapacidadDetalle: 'DiscapacidadDetalle modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo documento', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      documento: 'Documento modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo estadoCivilId', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { estadoCivilId: 2 })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo fechaNacimiento', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      fechaNacimiento: 'FechaNacimiento modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo genero', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { genero: 'Genero modificado' })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo telefonoFijoNumero', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      telefonoFijoNumero: 'TelefonoFijoNumero modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo telefonoFijoPrefijo', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, {
      telefonoFijoPrefijo: 'TelefonoFijoPrefijo modificado',
    })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo tieneLicenciaConducir', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { tieneLicenciaConducir: false })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo tieneMovilidadPropia', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { tieneMovilidadPropia: false })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })

  it('should completar req con solo tipoDocumentoId', async () => {
    const datosPostulanteLogueadoMock = require('./datos-postulante-logueado.mock')

    jest.spyOn(service, 'getDatosDePostulante').mockImplementation(() => Promise.resolve(datosPostulanteLogueadoMock))

    const result = await service.getDatosDePostulanteReqCoreDto({} as RequestContext, { tipoDocumentoId: 2 })

    expect(service.getDatosDePostulante).toHaveBeenCalledWith({} as RequestContext)

    expect(result).toMatchSnapshot()
  })
})
