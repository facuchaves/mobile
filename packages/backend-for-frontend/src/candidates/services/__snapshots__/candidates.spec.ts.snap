// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidates  should completar req con solo apellido 1`] = `
Object {
  "apellido": "Apellido modificado",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo celularNumero 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "CelularNumero modificado",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo celularPrefijo 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "CelularPrefijo modificado",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo discapacidadDetalle 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": "DiscapacidadDetalle modificado",
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo documento 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "Documento modificado",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo email 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "Email modificado",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo estadoCivilId 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 2,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo fechaNacimiento 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "FechaNacimiento modificado",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo genero 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "Genero modificado",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo nombre 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Nombre modificado",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo paisNacimientoId 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 2,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo telefonoFijoNumero 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "TelefonoFijoNumero modificado",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo telefonoFijoPrefijo 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "TelefonoFijoPrefijo modificado",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo tieneLicenciaConducir 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": false,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo tieneMovilidadPropia 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": false,
  "tipoDocumentoId": 45,
}
`;

exports[`Candidates  should completar req con solo tipoDocumentoId 1`] = `
Object {
  "apellido": "Barragán",
  "celularNumero": "64738023",
  "celularPrefijo": "5415",
  "discapacidadDetalle": null,
  "documento": "31034553",
  "email": "abarragan@navent.com",
  "estadoCivilId": 1,
  "fechaNacimiento": "19-08-1983",
  "genero": "masculino",
  "nombre": "Adrián Guillermo",
  "paisNacimientoId": 1,
  "telefonoFijoNumero": "35284312",
  "telefonoFijoPrefijo": "5411",
  "tieneLicenciaConducir": null,
  "tieneMovilidadPropia": null,
  "tipoDocumentoId": 2,
}
`;
